core/field_number.js

```diff
@@ -151,6 +154,7 @@ Blockly.FieldNumber.prototype.setConstraints_ = function(opt_min, opt_max,
       (Math.floor(opt_precision) != opt_precision);
   this.negativeAllowed_ = (typeof opt_min == 'undefined') || isNaN(opt_min) ||
       opt_min < 0;
+  this.exponentialAllowed_ = this.decimalAllowed_;
```

Comment:
Not 110% sure it makes sense to use a flag for this (`exponentialsAllowed_`) like with the other values, but it's here for code consistency. I made it equal to `decimalAllowed_` because if you *can't* input decimals, you shouldn't be able to input `e` either, so that you can't write (for example) `7e-1`, which is effectively a decimal value (0.7).